merkle/compact/tree.go

```diff
@@ -237,8 +238,8 @@ func (t *Tree) AddLeafHash(leafHash []byte, setNodeFn setNodeFunc) (int64, error
 		return assignedSeq, nil
 	}
 
-	// Initialize our running hash value to the leaf hash.
-	hash := leafHash
+	// Initialize our running hash value to a copy of the leaf hash
+	hash := append(make([]byte, 0, len(leafHash)), leafHash...)
```

Comment:
`append([]byte(nil), leafHash...)`

Reply:
While it would work the same I think using an empty slice is slightly clearer than a nil one.

Reply:
Dunno, leaving up to to. To me the shorted the better. Also, consider [this](https://github.com/go101/go101/wiki/How-to-perfectly-clone-a-slice%3F) article, there is even a third way apparently.